pkg/providers/aws/cluster_vpc.go

```diff
@@ -0,0 +1,397 @@
+	if err != nil {
+		return errorUtil.Wrap(err, "error building subnet group name")
+	}
+
+	// get cluster cidr group
+	vpcID, cidr, err := GetCidr(ctx, c, ec2Svc)
+	if err != nil {
+		return errorUtil.Wrap(err, "error finding cidr block")
+	}
+
+	foundSecGroup, err := getSecurityGroup(ec2Svc, secName)
+	if err != nil {
+		return errorUtil.Wrap(err, "error get security group")
+	}
+
+	if foundSecGroup == nil {
+		// create security group
+		logrus.Info(fmt.Sprintf("creating security group from cluster %s", clusterID))
+		if _, err := ec2Svc.CreateSecurityGroup(&ec2.CreateSecurityGroupInput{
+			Description: aws.String(fmt.Sprintf("security group for cluster %s", clusterID)),
+			GroupName:   aws.String(secName),
+			VpcId:       aws.String(vpcID),
+		}); err != nil {
+			return errorUtil.Wrap(err, "error creating security group")
+		}
+		return nil
+	}
+
+	// build ip permission
+	ipPermission := &ec2.IpPermission{
+		IpProtocol: aws.String("-1"),
+		IpRanges: []*ec2.IpRange{
+			{
+				CidrIp: aws.String(cidr),
+			},
+		},
+	}
+
+	// check if correct permissions are in place
+	for _, perm := range foundSecGroup.IpPermissions {
+		if reflect.DeepEqual(perm, ipPermission) {
+			logrus.Info("ip permissions are correct for postgres resource")
+			return nil
+		}
+	}
+
+	// authorize ingress
+	logrus.Info(fmt.Sprintf("setting ingress ip permissions for %s ", *foundSecGroup.GroupName))
+	if _, err := ec2Svc.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: aws.String(*foundSecGroup.GroupId),
+		IpPermissions: []*ec2.IpPermission{
+			ipPermission,
+		},
+	}); err != nil {
+		return errorUtil.Wrap(err, "error authorizing security group ingress")
+	}
+
+	return nil
+}
+
+// GetVPCSubnets returns a list of subnets associated with cluster VPC
+func GetVPCSubnets(ctx context.Context, c client.Client, ec2Svc ec2iface.EC2API) ([]*ec2.Subnet, error) {
+	logrus.Info("gathering cluster vpc and subnet information")
+	// poll subnets to ensure credentials have reconciled
+	subs, err := getSubnets(ec2Svc)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting subnets")
+	}
+
+	// get cluster vpc
+	foundVPC, err := getVpc(ctx, c, ec2Svc)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "error getting vpcs")
+	}
+
+	// check if found cluster vpc
+	if foundVPC == nil {
+		return nil, errorUtil.New("error, unable to find a vpc")
+	}
+
+	// find associated subnets
+	var associatedSubs []*ec2.Subnet
+	for _, sub := range subs {
+		if *sub.VpcId == *foundVPC.VpcId {
+			associatedSubs = append(associatedSubs, sub)
+		}
+	}
+
+	// check if found subnets associated with cluster vpc
+	if associatedSubs == nil {
+		return nil, errorUtil.New("error, unable to find subnets associated with cluster vpc")
+	}
+
+	return associatedSubs, nil
+}
+
+// GetSubnetIDS returns a list of subnet ids associated with cluster vpc
+func GetPrivateSubnetIDS(ctx context.Context, c client.Client, ec2Svc ec2iface.EC2API) ([]*string, error) {
```

Comment:
could we pass in the clusterId here or rename the func to be more descriptive? (e.g. GetPrivateSubnetIDSForCluster)